src/ethereum_test_fixtures/file.py

```diff
@@ -65,10 +66,16 @@ def collect_into_file(self, file_path: Path):
         add the hash to the info field on per-fixture basis.
         """
         json_fixtures: Dict[str, Dict[str, Any]] = {}
-        for name, fixture in self.items():
-            json_fixtures[name] = fixture.json_dict_with_info()
-        with open(file_path, "w") as f:
-            json.dump(json_fixtures, f, indent=4)
+        lock_file_path = file_path.with_suffix(".lock")
+        with FileLock(lock_file_path):
+            if file_path.exists():
+                with open(file_path, "r") as f:
+                    json_fixtures = json.load(f)
+            for name, fixture in self.items():
+                json_fixtures[name] = fixture.json_dict_with_info()
+
+            with open(file_path, "w") as f:
+                json.dump(dict(sorted(json_fixtures.items())), f, indent=4)
```

Comment:
I would prefer to have the fixtures ordered as pytest executes them with xdist disabled. But I'm guessing that you added `sorted` as it's no longer possible and otherwise the ordering with xdist is now non-deterministic?

One way to maintain this would be to apply sorted on test id per test function in `pytest_collection_modifyitems`.

Reply:
> One way to maintain this would be to apply sorted on test id per test function in `pytest_collection_modifyitems`.

Even then I think it's not possible to know which worker is going to finish filling first, even if the workload is sent out in order.

The reason why I thought of adding this is because it makes it a bit easier to visually compare the output fixtures and the whole fixture file is already in memory so it should not be that much extra overhead.

Reply:
I think the main thing is to ensure that the behavior of "xdist" vs "no xdist" is consistent, which it now will be as `sorted` is applied in either case. I don't mind changing the behavior, thanks for the follow-up!